shm/inc/Cargo.hpp

```diff
@@ -4,15 +4,22 @@
 
 class Cargo {
 public:
-    Cargo(const std::string&, size_t, size_t);
+    Cargo(const std::string& name, size_t amount, size_t basePrice_) 
+        : name_(name)
+        , amount_(amount)
+        , basePrice_(basePrice)
+    {}
 
-    Cargo& operator+=(const size_t);
-    Cargo& operator-=(const size_t);
-    bool operator==(const Cargo&) const;
+    virtual ~Cargo();
+
+    virtual Cargo& operator+=(const size_t) = 0;
+    virtual Cargo& operator-=(const size_t) = 0;
+    virtual bool operator==(const Cargo&) const = 0;
+
+    virtual std::string getName() const = 0;
+    virtual size_t getAmount() const = 0;
+    virtual size_t getBestPrice() const = 0;
 
-    std::string getName() const { return name_; }
-    size_t getAmount() const { return amount_; }
-    size_t getBestPrice() const { return basePrice_; }
 
 private:
```

Comment:
You should change private to protected, because we will don't have permission in derived classes